src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -309,6 +309,14 @@ public class SqlMutationEngine : IMutationEngine
         public async Task<IActionResult?> ExecuteAsync(RestRequestContext context)
         {
             Dictionary<string, object?> parameters = PrepareParameters(context);
+            string roleName = GetHttpContext().Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER];
```

Comment:
Can move the block from line 312-319 to the `ELSE` block since for delete operation, we don't need this info.

Reply:
Updated

Reply:
Updated to calculate all of this only for other operations.